src/Microsoft.AspNet.Mvc.Core/ModelBinding/Validation/DefaultObjectValidator.cs

```diff
@@ -166,32 +152,40 @@ private void MarkPropertiesAsSkipped(string currentModelKey, ModelMetadata metad
             }
         }
 
-        private bool ValidateProperties(
+        private IList<IModelValidator> GetValidators(IModelValidatorProvider provider, ModelMetadata metadata)
+        {
+            var validatorProviderContext = new ModelValidatorProviderContext(metadata);
+            provider.GetValidators(validatorProviderContext);
+            return validatorProviderContext.Validators;
+        }
+
+        private bool ValidateChildNodes(
             string currentModelKey,
             ModelExplorer modelExplorer,
             ValidationContext validationContext)
         {
             var isValid = true;
+            var validators = ExpandValidationNodeAndGetValidators(validationContext, modelExplorer);
 
-            foreach (var property in modelExplorer.Metadata.Properties)
+            foreach (var childNode in validationContext.ValidationNode.ChildNodes)
             {
-                var propertyExplorer = modelExplorer.GetExplorerForProperty(property.PropertyName);
-                var propertyMetadata = propertyExplorer.Metadata;
+                var childModelExplorer = childNode.ModelMetadata.MetadataKind == Metadata.ModelMetadataKind.Type ?
+                    _modelMetadataProvider.GetModelExplorerForType(childNode.ModelMetadata.ModelType, childNode.Model) :
+                    modelExplorer.GetExplorerForProperty(childNode.ModelMetadata.PropertyName);
```

Comment:
IIRC the problem with `new ModelExplorer(...)` was DataAnnotationsValidator needs a container to operate. if we blindly do new here it would break the hierarchy. I agree that we should probably yank out ModelExplorer but that is another PR.